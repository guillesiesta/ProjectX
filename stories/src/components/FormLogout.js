import React, { Component } from 'react';

class FormLogout extends Component {

  constructor(props) {
    super(props);
    this.state = {username: '', password: ''};
    this.handleNameEmpty = this.handleNameEmpty.bind(this);
  }

  handleNameEmpty() {
      //this.setState({username:''});
      console.log("Username de logout "+this.state.username)
      this.setState({username:''});
      this.setState({password:''});
      this.props.getUsername(this.state.username);
  }

  render(){
    return(
      <form >
        <input type="button" value="Logout" onClick={this.handleNameEmpty}/>
      </form>
    );
  }

}

export default FormLogout;
