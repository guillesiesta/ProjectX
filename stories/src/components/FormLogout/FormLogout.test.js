import React, {Component} from 'react';
import { shallow } from 'enzyme';

import FormLogout from './FormLogout';

describe('FormLogout', ()=> {
  test('Rendering the simple component', ()=>{

    const wrapper = shallow(
      <FormLogout />
    );

    expect(wrapper).toMatchSnapshot();
  })
})
