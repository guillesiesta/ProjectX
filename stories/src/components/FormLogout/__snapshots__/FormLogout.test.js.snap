// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLogout Rendering the simple component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormLogout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        onClick={[Function]}
        type="button"
        value="Logout"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "onClick": [Function],
        "type": "button",
        "value": "Logout",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          onClick={[Function]}
          type="button"
          value="Logout"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onClick": [Function],
          "type": "button",
          "value": "Logout",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
